src/test/java/com/amazon/opendistroforelasticsearch/sql/unittest/NestedFieldProjectionTest.java

```diff
@@ -239,10 +240,290 @@ public void multipleDifferentNestedFields() {
+                        boolQuery(
+                            must(
+                                nestedQuery(
+                                    deepNestedQuery(
+                                        deepSource(
+                                            nestedQuery(
+                                                path("projects.address"),
+                                                innerHits("projects.address.*")
+                                            )
+                                        )
+                                    ),
+                                    path("projects"),
+                                    innerHits("projects.*")
+                                )
+                            )
+                        )
+                    )
+                )
+            )
+        );
+    }
+
+    @Test
+    public void regularAndDeeplyNestedFieldInSelect() {
+        assertThat(
+            query("SELECT title, nested(projects.address.city) from employee_nested_2"),
+            source(
+                boolQuery(
+                    filter(
+                        boolQuery(
+                            must(
+                                nestedQuery(
+                                    path("projects.address"),
+                                    innerHits("projects.address.city")
+                                )
+                            )
+                        )
+                    )
+                ),
+                fetchSource("title")
+            )
+        );
+    }
+
+    @Test
+    public void deepSelectNestedField() {
+        assertThat(
+            query("SELECT nested(projects.address.state), nested(projects.address.city), " +
+                    "nested(projects.started_year), nested(projects.name) FROM employees_nested_2"),
+            source(
+                boolQuery(
+                    filter(
+                        boolQuery(
+                                must(
+                                    nestedQuery(
+                                        deepNestedQuery(
+                                            deepSource(
+                                                nestedQuery(
+                                                    path("projects.address"),
+                                                    innerHits("projects.address.state", "projects.address.city")
+                                                )
+                                            )
+                                        ),
+                                        path("projects"),
+                                        innerHits("projects.started_year", "projects.name")
+                                    )
+                                )
+                        )
+                    )
+                )
+            )
+        );
+    }
+
+    @Test
+    @Ignore("Deep query with WHERE condition is not fully supported")
```

Comment:
Thanks for adding the tests for the unsupported cases.

Reply:
No problem